text_mining.py

```diff
@@ -0,0 +1,201 @@
+"""
+text_mining.py
+Tatiana Anthony
+* Pulls 100 comments from /r/all, /r/popular (for defaults), the 15 largest 
+subreddits as of Thursday Febrary 23, 2017 at 9:30 AM, and a few subreddits I
+was interested in.
+* Pickles them to save the comments and scores of the comments.
+* Runs sentiment analysis on the comments
+* Scales sentiment by the score of the comment
+    - A positive comment that is upvoted will increase the subreddit sentiment
+    - A negative comment that is upvoted will decrease the subreddit sentiment
+    - A positive comment that is downvoted will decrease the subreddit sentiment
+    - A negative comment that is downvoted will increase the subreddit sentiment
+* Graphs the sentiments of each subreddit.
+"""
+import praw
+from praw.models import MoreComments
+from os.path import exists
+import pickle
+import numpy as np
+import matplotlib.pyplot as plt
+from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
+
+defaults = 'announcements+Art+AskReddit+askscience+aww+blog+books+creepy+dataisbeautiful'
+
+def read_key(name):
+    """
+    Inputs:  name of the read key file
+    Outputs:  The key
+    """
+    return open(name + '.txt').read().strip()
```

Comment:
I think easier way to store your secret key would be creating a separate python file in which a key is stored as a python variable. In this way, you can just import that python file and get an access to those variables.